share/completions/istioctl.fish

```diff
@@ -0,0 +1,176 @@
+# fish completion for istioctl                             -*- shell-script -*-
+
+function __istioctl_debug
+    set -l file "$BASH_COMP_DEBUG_FILE"
+    if test -n "$file"
+        echo "$argv" >> $file
+    end
+end
+
+function __istioctl_perform_completion
+    __istioctl_debug "Starting __istioctl_perform_completion"
+
+    # Extract all args except the last one
+    set -l args (commandline -opc)
+    # Extract the last arg and escape it in case it is a space
+    set -l lastArg (string escape -- (commandline -ct))
+
+    __istioctl_debug "args: $args"
+    __istioctl_debug "last arg: $lastArg"
+
+    set -l requestComp "$args[1] __complete $args[2..-1] $lastArg"
+
+    __istioctl_debug "Calling $requestComp"
+    set -l results (eval $requestComp 2> /dev/null)
+
+    # Some programs may output extra empty lines after the directive.
+    # Let's ignore them or else it will break completion.
+    # Ref: https://github.com/spf13/cobra/issues/1279
+    for line in $results[-1..1]
+        if test (string trim -- $line) = ""
+            # Found an empty line, remove it
+            set results $results[1..-2]
+        else
+            # Found non-empty line, we have our proper output
+            break
+        end
+    end
+
+    set -l comps $results[1..-2]
+    set -l directiveLine $results[-1]
+
+    # For Fish, when completing a flag with an = (e.g., <program> -n=<TAB>)
+    # completions must be prefixed with the flag
+    set -l flagPrefix (string match -r -- '-.*=' "$lastArg")
+
+    __istioctl_debug "Comps: $comps"
+    __istioctl_debug "DirectiveLine: $directiveLine"
+    __istioctl_debug "flagPrefix: $flagPrefix"
+
+    for comp in $comps
+        printf "%s%s\n" "$flagPrefix" "$comp"
```

Comment:
This can just be

```fish
printf '%s\n' $flagPrefix$comp
```

No need for the for-loop.

Reply:
actually the loop _is_ needed here. If I replace it with a single `printf` it won't complete all options and only ever complete to the first in the list

Reply:
> actually the loop is needed here

I don't think so. Probably you just need to quote flagPrefix since it might be empty (?)

`printf '%s\n' "$flagPrefix"$comp`